third_party/terraform/utils/node_config.go.erb

```diff
@@ -502,4 +509,58 @@ func flattenSandboxConfig(c *containerBeta.SandboxConfig) []map[string]interface
 	}
 	return result
 }
+
+func containerNodePoolLabelsSuppress(k, old, new string, d *schema.ResourceData) bool {
+	o, n := d.GetChange("node_config.0.sandbox_config.0.sandbox_type")
+	if o == nil || n == nil {
+		return false
+	}
+
+	if strings.HasPrefix(k, "node_config.0.labels.sandbox.gke.io/") {
```

Comment:
I'm not convinced this will work. It would be worth verifying (with some print statements or a test) but I'm pretty sure 'k' here is going to be the full list, rather than this func being called on each element of the list.

Reply:
So the behavior here is really interesting. You actually get each key and value under the root independently and can return a different result for each. For example, dumping the `k`, `old`, and `new` values for each invocation you'd see something like:

```
k=node_config.0.labels.# old=2 new=1
k=node_config.0.labels.test.terraform.io/test old=foo new=foo
k=node_config.0.labels.test.sandbox.gke.io/runtime old=gvisor new=
```

It turns out to not be important (see my next comment) now, but the initial pass did depend on this admittedly strange behavior. I think it has to do with `old` and `new` being strings.